### PR TITLE
ci: add workflow to automatically merge main into v2 branch

### PR DESCRIPTION
**Pull request by MartinMinkov**

### CODE REVIEW

File: .github/workflows/auto-merge-main-to-v2.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  push:
+    branches:
+      - main
+
+jobs:
+  auto-merge:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
Comment (shimkiv) [inline .github/workflows/auto-merge-main-to-v2.yml:25]: Please update this action to use the latest (`v4` I think).

File: .github/workflows/auto-merge-main-to-v2.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Try to merge main into v2
+        run: |
+          git checkout v2
+          git merge origin/main
+          git push origin v2
+        continue-on-error: true
+
+      - name: Create Pull Request if merge failed
+        if: failure()
+        uses: peter-evans/create-pull-request@v3
```
Comment (shimkiv) [inline .github/workflows/auto-merge-main-to-v2.yml:43]: Please update this action to use the latest (`v6` I think).

File: .github/workflows/auto-merge-main-to-v2.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        with:
+          branch: auto-merge-main-to-v2
```
Comment (shimkiv) [inline .github/workflows/auto-merge-main-to-v2.yml:45]: > Commit all changes to a new branch, or update an existing pull request branch.

Are concerned about the edge cases when two different merges into the `main` branch will cause this workflow to kick off and they both will push to the same `auto-merge-main-to-v2` branch with possible conflicts?
Or `v2` is not supposed to last long enough to spend time on that kind of concerns?

**Review by shimkiv — APPROVED** [review]

File: .github/workflows/auto-merge-main-to-v2.yml (hunk 3 above)
Comment (MartinMinkov) [reply .github/workflows/auto-merge-main-to-v2.yml]: That's a great point! I added this [slight change here ](https://github.com/o1-labs/o1js/pull/1744/commits/15fe7d699833fa44593e09df0d4d2fbe0c09f327)to make `auto-merge-main-to-v2` have a postfix of the git commit. That should fix the issues with two merges clobbering each other.

Comment (shimkiv) [reply .github/workflows/auto-merge-main-to-v2.yml]: Thank you!

**Pull request merged**